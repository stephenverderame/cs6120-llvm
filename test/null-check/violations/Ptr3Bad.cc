#include <cstdio>
#include <cstdlib>
// NOLINTBEGIN
int main()
{
    int*** p = (int***)malloc(sizeof(int***));
    if (p) {
        *p = (int**)malloc(sizeof(int**));
        if (*p) {
            **p = (int*)malloc(sizeof(int*));
            if (**p) {
                ***p = 0;
            }
            printf("%d\n", ***p);
        }
    }
}
// NOLINTEND