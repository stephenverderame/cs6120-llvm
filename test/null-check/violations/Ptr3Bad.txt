[31mSafety Violation[00m: Use of potentially null pointer in load at Ptr3Bad.cc:14
clang++-17: error: clang frontend command failed with exit code 2 (use -v to see invocation)
