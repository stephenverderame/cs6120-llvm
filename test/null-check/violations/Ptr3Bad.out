[31mSafety Violation[00m: Use of potentially null pointer in load at Ptr3Bad.cc:14
